homeassistant/components/tedee/coordinator.py

```diff
@@ -0,0 +1,110 @@
+"""Coordinator for Tedee locks."""
+import asyncio
+from datetime import timedelta
+import logging
+import time
+
+from pytedee_async import (
+    TedeeAuthException,
+    TedeeClient,
+    TedeeClientException,
+    TedeeDataUpdateException,
+    TedeeLocalAuthException,
+    TedeeLock,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_LOCAL_ACCESS_TOKEN, DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=20)
+STALE_DATA_INTERVAL = 300
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TedeeApiCoordinator(DataUpdateCoordinator[dict[int, TedeeLock]]):
+    """Class to handle fetching data from the tedee API centrally."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize coordinator."""
+        self.tedee_client = TedeeClient(
+            personal_token=entry.data.get(CONF_ACCESS_TOKEN),
+            local_token=entry.data.get(CONF_LOCAL_ACCESS_TOKEN),
+            local_ip=entry.data.get(CONF_HOST),
+        )
+        self._initialized = False
+        self._next_get_locks = time.time()
+        self._last_data_update = time.time()
+        self._stale_data = False
+        super().__init__(
```

Comment:
In this function `self.config_entry` will be set to the current entry, so if you move this function to the top you can remove the `entry` parameter of this class